### PR TITLE
Update getNavLink helper for about page routing

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: app-project

Describe your changes:
Partly toward #2061. I've updated the `getNavLink` helper to build the href for the about page to use an absolute url to zooniverse.org when in production. I also updated it to route to `/about/research`,  but we'll still  need the redirect as described in #2061 for direct links. When we're ready to launch the about pages, this change to use an absolute url in production in the helper will be reverted.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by srallen** [comment]
This can be tested by using `yarn dev` for the staging environment, it should still use relative links, then`yarn start` for production where the link will be absolute.

File: packages/app-project/src/components/ProjectHeader/helpers/getNavLinks/getNavLinks.js [hunk 1]
```diff
@@ -2,9 +2,10 @@ import counterpart from 'counterpart'
 
 function getNavLinks (isLoggedIn, baseUrl, defaultWorkflow) {
   const classifyHref = defaultWorkflow ? `${baseUrl}/classify/workflow/${defaultWorkflow}` : `${baseUrl}/classify`
+  const aboutHref = (process.env.PANOPTES_ENV === 'production') ? `https://www.zooniverse.org${baseUrl}/about` : `${baseUrl}/about/research`
```
Comment (eatyourgreens) [inline packages/app-project/src/components/ProjectHeader/helpers/getNavLinks/getNavLinks.js:5]: Staging uses the production API so I think you want to test `APP_ENV` here. At least, I think that's what we called the deployment environment variable.

File: packages/app-project/src/components/ProjectHeader/helpers/getNavLinks/getNavLinks.spec.js [hunk 2]
```diff
@@ -60,7 +62,44 @@ describe('Helper > getNavLinks', function () {
...
+        }
+      ])
+    })
+  })
+
+  describe('when PANOPTES_ENV is production', function () {
+    let previousEnv
+    before(function () {
+      previousEnv = process.env.PANOPTES_ENV
+      process.env.PANOPTES_ENV = 'production'
```
Comment (eatyourgreens) [inline packages/app-project/src/components/ProjectHeader/helpers/getNavLinks/getNavLinks.spec.js:92]: In other tests, we use `proxyquire` to mock the Node environment.
https://github.com/zooniverse/front-end-monorepo/blob/debb99711151ce9a1e28d56790bd3a2fe64b44f2/packages/app-project/server/set-logging.spec.js#L12-L20

File: packages/app-project/src/components/ProjectHeader/helpers/getNavLinks/getNavLinks.js (hunk 1 above)
Comment (srallen) [reply packages/app-project/src/components/ProjectHeader/helpers/getNavLinks/getNavLinks.js:5]: `APP_ENV` is used for sentry: https://github.com/zooniverse/front-end-monorepo#environment-variables

Comment (srallen) [reply packages/app-project/src/components/ProjectHeader/helpers/getNavLinks/getNavLinks.js:5]: I'd like the QA to reflect what we're actually gonna deploy to production for now. When we're ready to deploy the about pages, we can revert it.

**Review by goplayoutside3 — APPROVED** [review]

**Pull request merged**